logging/src/lib.rs

```diff
@@ -0,0 +1,81 @@
+//! Implementation of an asynchronous logger for `lib3h`
+
+use chrono;
+pub use slog_async;
+pub use slog_scope::{crit, debug, error, info, trace, warn};
+pub use slog_term;
+
+use slog::Drain;
+pub use slog::{slog_crit, slog_debug, slog_error, slog_info, slog_o, slog_trace, slog_warn};
+
+// pub struct Logger {
+//     _global_logger_guard: slog_scope::GlobalLoggerGuard,
+// }
+/// This struct is used as a wrapper around the `slog` crate for our customized logger.
+/// This has to be instantiate at the top of the user program in order to keep
+/// the logging thread alive.
+pub struct Logger(slog_scope::GlobalLoggerGuard);
+
+impl Logger {
+    pub fn new() -> Self {
+        Logger::term_logger()
+    }
+
```

Comment:
If this is a singleton it probably shouldn't be named `new()`. Is it setup / teardown style? Like an `init()` method? Or more like a wrapper function around a singleton? Eg. `get()` or `instance()` or `try...()`?

Reply:
I wanted to stick to idiomatic Rust with a `new` constructor, but indeed, calling a `init()` is probably more accurate, thanks for your input on this.